src/assets/scripts/client/airline/AirlineModel.js

```diff
@@ -152,34 +158,134 @@ export default class AirlineModel extends BaseModel {
     }
 
     /**
+     * Resets the current list of `flightNumbersInUse`.
+     *
+     * This can be used when changing airports and all existing
+     * aircraft are removed.
+     *
+     * @for AirlineModel
+     * @method reset
+     */
+    reset() {
+        this.flightNumbersInUse = [];
+    }
+
+    /**
+     * @method getRandomAircraftType
+     * @param fleet {string}
+     * @return {string}
+     */
+    getRandomAircraftType(fleet = '') {
+        if (fleet === '') {
+            return this._getRandomAircraftTypeFromAllFleets();
+        }
+
+        return this._getRandomAircraftTypeFromFleet(fleet);
+    }
+
+    // TODO: the logic here can be simplified.
+    /**
+     * Create a flight number/identifier
+     *
+     * @for AirlineModel
+     * @method generateFlightNumber
+     * @return flightNumber {string}
+     */
+    generateFlightNumber() {
+        let flightNumber = '';
+        let list = '0123456789';
+
+        // Start with a number other than zero
+        flightNumber += choose(list.substr(1));
+
+        if (this.flightNumberGeneration.alpha) {
+            // TODO: why `this.flightNumberGeneration.length - 3`?  enumerate the magic number.
+            for (let i = 0; i < this.flightNumberGeneration.length - 3; i++) {
+                flightNumber += choose(list);
+            }
+
+            list = 'abcdefghijklmnopqrstuvwxyz';
+
+            for (let i = 0; i < 2; i++) {
+                flightNumber += choose(list);
+            }
```

Comment:
this `if` block is all about N number registration callsigns, eg "N123AB".

the first `for` loop generates three random numbers, and the second `for` loop generates two random letters. Eventually the callsign generation could use a minor refactor to allow more rules about how to shape the callsigns... number ranges, or a template, such as `###A` to allow `543D`, `180A`, etc. Of next to zero importance though. Anyway, it was the application of N117KS that led to the magic and undocumented numbers.